README.md

```diff
@@ -522,60 +522,133 @@ You can customize the authentication process at different levels:
-    console.log('State just changed to:', this.auth.getState());
+The [authentication process](https://api.pryv.com/reference/#authenticate-your-app) implementation on the frontend can be defined in the following states:
 
-    let text = this.auth.pryvService.defaultButtonMessage();
-    document.querySelector(`#${this.auth.settings.spanButtonID}`).innerText = text;
+1. Loading: while the visual assets are loading
+2. Initialized: visual l
+3. Need sign in: 
+4. Authorized
+5. Sign out
+6. Error
+
+```javascript
+async onStateChange (state) {
+  switch (state.status) {
+    case AuthStates.LOADING:
+      this.text = getLoadingMessage(this);
+      break;
+    case AuthStates.INITIALIZED:
+      this.text = getInitializedMessage(this, this.serviceInfo.name);
+      break;
+    case AuthStates.NEED_SIGNIN:
+      const loginUrl = state.authUrl || state.url; // .url is deprecated
+      if (this.authSettings.authRequest.returnURL) { // open on same page (no Popup) 
+        location.href = loginUrl;
+        return;
+      } else {
+        startLoginScreen(this, loginUrl);
+      }
+      break;
+    case AuthStates.AUTHORIZED:
+      this.text = state.username;
+      this.saveAuthorizationData({
+        apiEndpoint: state.apiEndpoint,
+        username: state.username
+      });
+      break;
+    case AuthStates.LOGOUT:
+      const message = this.messages.LOGOUT_CONFIRM ? this.messages.LOGOUT_CONFIRM : 'Logout ?';
+      if (confirm(message)) {
+        this.deleteAuthorizationData();
+        this.auth.init();
+      }
+      break;
+    case AuthStates.ERROR:
+      this.text = getErrorMessage(this, state.message);
+      break;
+    default:
+      console.log('WARNING Unhandled state for Login: ' + state.status);
   }
+  if (this.loginButtonText) {
+    this.loginButtonText.innerHTML = this.text;
+  }
+}
+```
+
+##### Button actions
+
+The button actions should be handled by the AuthController in the following way:
+
+```javascript
+// LoginButton.js
+onClick () {
+  this.auth.handleClick();
 }
+```
 
-onClick(button) {
-  console.log('My custom on click event. Currect auth state is: ', this.auth.getState());
-  if (button.auth.getState().id === Pryv.Browser.AuthStates.AUTHORIZED) {
-    button.auth.logOut();
-  } else if (button.auth.getState().id === Pryv.Browser.AuthStates.INITIALIZED) {
-    button.auth.pryvService.startAuthRequest(button.auth);
-    const authUrl = button.auth.pryvService.getAccessData().authUrl;
-    // open the authUrl in a new window or web view
-    window.open(authUrl, 'You custom Sign-in');
+```javascript
+// AuthController.js
+async handleClick () {
+  if (isAuthorized.call(this)) {
+    this.state = { status: AuthStates.LOGOUT };
+  } else if (isInitialized.call(this)) {
+    this.startAuthRequest();
+  } else if (isNeedSignIn.call(this)) {
+    // reopen popup
+    this.state = this.state;
+  } else {
+    console.log('Unhandled action in "handleClick()" for status:', this.state.status);
```

Comment:
Maybe worth it to handle error scenario - if person rejected, he can click and sign again, what do you think?

Reply:
done